src/main/java/com/manardenza/service/ReservationService.java

```diff
@@ -0,0 +1,57 @@
+package com.manardenza.service;
+
+import com.manardenza.dao.ReservationDaoImpl;
+import com.manardenza.entity.Hotel;
+import com.manardenza.entity.Reservation;
+import com.manardenza.entity.Room;
+import com.manardenza.login.CurrentUser;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public final class ReservationService {
+
+    private static ReservationService instance;
+    private ReservationDaoImpl reservationDao;
+    private CurrentUser currentUser;
+
+    public ReservationService(ReservationDaoImpl reservationDao) {
+        this.reservationDao = reservationDao;
+    }
+
+    public Long bookRoom(Date reservedFrom, Date reservedTo, Room room, Hotel hotel) {
+        Reservation newReservation = new Reservation(reservedFrom, reservedTo, currentUser.getUser(), room, hotel);
+        reservationDao.save(newReservation);
+
+        return newReservation.getId();
+    }
+
+    public boolean cancelReservation(long idReservation) {
+        return reservationDao.getAll().removeIf(p -> p.getId() == idReservation);
```

Comment:
You cannot do like this - it will not be persisted. You should use delete() method